testing/test_terminal.py

```diff
@@ -327,16 +327,17 @@ def test_rewrite(self, pytester: Pytester, monkeypatch) -> None:
         tr.rewrite("hey", erase=True)
         assert f.getvalue() == "hello" + "\r" + "hey" + (6 * " ")
 
+    @pytest.mark.parametrize("category", ["foo", "failed", "error", "passed"])
     def test_report_teststatus_explicit_markup(
-        self, monkeypatch: MonkeyPatch, pytester: Pytester, color_mapping
+        self, monkeypatch: MonkeyPatch, pytester: Pytester, color_mapping, category: str
     ) -> None:
         """Test that TerminalReporter handles markup explicitly provided by
         a pytest_report_teststatus hook."""
         monkeypatch.setenv("PY_COLORS", "1")
         pytester.makeconftest(
-            """
+            f"""
             def pytest_report_teststatus(report):
-                return 'foo', 'F', ('FOO', {'red': True})
+                return '{category}', 'F', ('FOO', {{'red': True}})
```

Comment:
It's probably safer to use a string repr for cases when it gets to have a single quote inside for whatever reason:
```suggestion
                return {category !r}, 'F', ('FOO', {{'red': True}})
```